lab/lab-guide.md

```diff
@@ -0,0 +1,802 @@
+
+In this hands-on lab, we will build serverless, AI RAG applications using .NET Aspire, Semantic Kernel and Azure Cosmos DB with DiskANN, vector search, and full-text search capabilities! You will learn how to implement Semantic Kernel OpenAI Plug-ins, NoSQL connectors, and semantic caching! Gain practical insights into how to design, build, deploy and scale RAG pattern applications in Azure! The skills in this lab will give you a solid foundation to create your own generative AI applications. Its recommended to have C# or other programming language experience before completing this lab.
+
+
+## What are we doing?
+
+This lab guides you through the steps to implement generative AI capabilities for a ASP.NET Core Blazor application with Azure Cosmos DB for NoSQL and Azure OpenAI service integration using Semantic Kernel and .NET Aspire. These are the main tasks you will accomplish in this lab.
+
+1. Connect to the application and explore the .NET Aspire dashboard. 
+1. Implement Semantic Kernel to coordinate sending user prompts to a Chat GPT model in Azure OpenAI Service and store the responses in Azure Cosmos DB.
+1. Implement and test a chat history feature to allow for more natural conversational interactions. 
+1. Implement RAG Pattern using custom data for augmented LLM responses.
+1. Implement and test a semantic cache for improved performance.
+
+
+# Prepare // TODO: environment contains pre-deployed resources, consider deleting from lab
+
+You're updating an existing .NET solution that has an ASP.NET Blazor application as its sole project. This project includes service classes using Azure Cosmos DB and Azure OpenAI that need to connect to the deployed services in Azure.
+
+Before moving to the next step, ensure you have completed the **Service Deployment** as well as the **Configuring a lab environment** in the [README File](../README.md).
+
+Once this is complete, you may begin the lab.
+
+
+# Exercise: Setup and run the starter web application // TODO: update me!
+
+The first step of this project is to ensure the existing starter application builds and runs successfully.
+
+There are a few requirements in this exercise:
+
+- Successfully build the application.
+- Run the application using the **Hot Reload** feature of .NET
+
+### Configure dev environment
+
+The [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) for Visual Studio Code requires [Docker](https://docs.docker.com/) to be installed on your local machine. The extension hosts the development container locally using the Docker host with the correct developer tools and dependencies preinstalled to complete this training module.
+
+This lab assumes the use of Visual Studio Code. However, you may use any IDE you wish.
+
+1. Open a local **Terminal or Shell**. Change the current working directory where you cloned this application, for example, **C:\Users\Mark\Repos\cosmosdb-chatgpt\src**.
```

Comment:
Change that url to something like, C:\Users\Mark\Repos\cosmosdb-nosql-copilot\src\ to reflect the new repo name.

Reply:
This whole section needs to be updated, didn't get to this one yet. Only added changes for three exercises so far (context window, RAG, semantic cache)